### ISSUE TITLE
Disable git prompt on a path in the gitfast plugin

### ISSUE THREAD

**Issue by TillLeiden** [ohmyzsh/ohmyzsh#8903]
Hi, I have the following problem. 
I want to disable the zsh vcs_info for a path, which I mount via ssh. I found a current workaround in https://github.com/sindresorhus/pure/issues/189#issuecomment-173844254

I added the following line to my .zshrc (at the top or bottom, same behavior)
```
zstyle ':vcs_info:*' disable-patterns "${(b)HOME}/foo(|/*)"
```
If I look at the output of `zstyle` I get also 
```
disable-patterns
        :vcs_info:* '/home/till/foo(|/*)'
```

this works fine for `plugins=(git)`, however I like to use the plugin `gitfast`. Here it seems to ignore the disable-patterns and still outputs the git information.

Do you have any suggestions?

Many thanks,
Till

**Comment by mcornella** [comment]
The gitfast plugin doesn't use `vcs_info`, it uses `git-prompt.sh`. You can disable the git prompt for that particular path in another way:
https://github.com/ohmyzsh/ohmyzsh/blob/b876198575cbf23d589ddc8da6b22254d3d5358b/plugins/gitfast/git-prompt.sh#L93-L96
https://github.com/ohmyzsh/ohmyzsh/blob/b876198575cbf23d589ddc8da6b22254d3d5358b/plugins/gitfast/git-prompt.sh#L385-L391

You have to set `$GIT_PS1_HIDE_IF_PWD_IGNORED` to a non-empty value and set up the path you want (`$HOME/foo`?) in such a way that `git check-ignore -q <path>` returns true; that is, you have to [globally ignore that directory](https://stackoverflow.com/a/39069087/5798232).

**Comment by TillLeiden** [comment]
Many thanks for the quick response.
As far as I understand, `git check-ignore -q <path>' only checks relative patterns from a repository (like .gitignore). So I cannot specify an absolute path to ignore mounted folders and all their subfolders.

I also added a few lines of code to check if the current path is a subpath of a given one
```
if [ "${PWD##/path/to/foo}" != "${PWD}" ]; then
	return $exit
fi
```
This works and no more automatic output from the gitfast plugin is displayed. But there is still some code running in the background, so I have to wait a few seconds until I can use the shell again in my mounted folders (e.g. after `cd`).
By the way, the git plugin with the use of `disable-patterns` seems to disable everything.

Then I added the above lines to disable the ``function git_prompt_info()`' of the `gitfast.plugin.zsh' file, and it works the way I need it to.

Thanks for your help and clarifications.

**Comment by TillLeiden** [comment]
Many thanks for the quick response.
As far as I understand, `git check-ignore -q <path>' only checks relative patterns from a repository (like .gitignore). So I cannot specify an absolute path to ignore mounted folders and all their subfolders.

I also added a few lines of code to check if the current path is a subpath of a given one
```
if [ "${PWD##/path/to/foo}" != "${PWD}" ]; then
	return $exit
fi
```
This works and no more automatic output from the gitfast plugin is displayed. But there is still some code running in the background, so I have to wait a few seconds until I can use the shell again in my mounted folders (e.g. after `cd`).
By the way, the git plugin with the use of `disable-patterns` seems to disable everything.

Then I added the above lines to disable the ``function git_prompt_info()`' of the `gitfast.plugin.zsh' file, and it works the way I need it to.

Thanks for your help and clarifications.
